services/web/server/src/simcore_service_webserver/users_handlers.py

```diff
@@ -109,3 +113,65 @@ async def delete_token(request: web.Request):
         raise web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
     except TokenNotFoundError as exc:
         raise web.HTTPNotFound(reason=f"Token for {service_id} not found") from exc
+
+
+# me/notifications -----------------------------------------------------------
+
+async def _get_user_notifications(redis_client: aioredis.Redis, user_id: int):
+    notifs = []
+    notif_wildcard = f'user_id={user_id}:notification_id=*'
+    async for scanned_notification_key in redis_client.scan_iter(match=notif_wildcard):
+        if notif_str := await redis_client.get(scanned_notification_key):
+            notif = json.loads(notif_str)
+            notifs.append(notif)
+    notifs.sort(key=lambda n: n["date"])
+    notifs.reverse()
+    return notifs
+
+
+@login_required
+async def get_user_notifications(request: web.Request):
+    redis_client = get_redis_user_notifications_client(request.app)
+    user_id = request[RQT_USERID_KEY]
+    notifs = await _get_user_notifications(redis_client, user_id)
+    # last 10 items only
+    return web.json_response(data={"data": notifs[-10:]})
+
+
+@login_required
+async def post_user_notification(request: web.Request):
+    redis_client = get_redis_user_notifications_client(request.app)
+    # body includes the new notification
+    notif = await request.json()
+    nid = random.randint(100, 1000)
```

Comment:
what is this id for? it does not appear very random. should it be a uuid? or is that just the index of the notification?

Reply:
done, it is a uuid now